service/follower/execution.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package follower
+
+import (
+	"fmt"
+
+	"github.com/gammazero/deque"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type Execution struct {
+	log     zerolog.Logger
+	queue   *deque.Deque
+	stream  RecordStreamer
+	records map[flow.Identifier]*Record
+}
+
+func NewExecution(log zerolog.Logger, stream RecordStreamer) *Execution {
+
+	s := Execution{
+		log:     log,
+		stream:  stream,
+		queue:   deque.New(),
+		records: make(map[flow.Identifier]*Record),
+	}
+
+	return &s
+}
+
+func (e *Execution) Update() (*ledger.TrieUpdate, error) {
+
+	// If we have updates available in the queue, let's get the oldest one and
+	// feed it to the indexer.
+	if e.queue.Len() != 0 {
```

Comment:
Nitpick, especially since the original code tends to panic on negative count, but I like when code like this reads something like `e.queue.Len() > 0`.

Reply:
It's actually a good practice to use `!=`, because it will compile to more efficient code.